### PR TITLE
[doc] wording, parameter inline modifier

### PR DESCRIPTION
**Pull request by robstoll**
Maybe this is only of a personal taste. When I first read "can be marked `inline`" then I thought about an annotation. Since there is `@inline` it could be overlooked. Thus I suggest we rephrase it to "can have an `inline` modifier".

I rewrote `final val` to `val` as far as I understood this will no longer be supported in the future. I am not sure if the examples are still valid without the `final` though

### CODE REVIEW

**Comment by robstoll** [comment]
Out of curiosity, what import is needed for `code"..."` (as shown in the `error` section) to work ?

**Comment by nicolasstucki** [comment]
```scala
import scala.compiletime.code
```

File: docs/docs/reference/metaprogramming/inline.md [hunk 1]
```diff
@@ -391,6 +394,8 @@ If an inline expansion results in a call `error(msgStr)` the compiler
 produces an error message containing the given `msgStr`.
 
 ```scala
+import scala.compiletime.error
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/inline.md]:
```suggestion
import scala.compiletime.{error, code}
```

File: docs/docs/reference/metaprogramming/inline.md [hunk 2]
```diff
@@ -360,18 +362,19 @@ Then:
...
+Match_ section above. Here is how `toIntT` can be defined:
 
 ```scala
 inline def toIntT[N <: Nat] <: Int = inline scala.compiletime.erasedValue[N] match {
   case _: Zero.type => 0
   case _: Succ[n] => toIntT[n] + 1
 }
 
-final val two = toIntT[Succ[Succ[Zero.type]]]
+val two = toIntT[Succ[Succ[Zero.type]]]
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/inline.md]:
```suggestion
final val two = toIntT[Succ[Succ[Zero.type]]]
```

File: docs/docs/reference/metaprogramming/inline.md [hunk 3]
```diff
@@ -360,18 +362,19 @@ Then:
   val dAny: None.type = defaultValue[Any]
 ```
 
-As another example, consider the type-level version of `toNat` above the we call
-`toIntT`: given a _type_ representing a Peano number, return the integer _value_
-corresponding to it. Consider the definitions of numbers as in the _Inline
-Match_ section aboce. Here's how `toIntT` can be defined:
+As another example, consider the type-level version of `toInt` below: 
+given a _type_ representing a [Peano number](https://wiki.haskell.org/Peano_numbers),
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/inline.md]:
```suggestion
given a _type_ representing a Peano number,
```

The Haskel version works on term rather that terms and types. This will be miseading to someone trying to understand the encoding.

**Comment by nicolasstucki** [comment]
Thanks @robstoll

**Comment by robstoll** [comment]
@nicolasstucki thanks for your time, next time I won't change something I am not sure about (e.g. remove `final`)
